# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible.cpp
# Commit: 09de143215a1
# Full Hash: 09de143215a13a22dcb2cd53d4ed1dfb60480c82
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-09-30 02:48 +0300
# Regressor Bug: 1987945
# File Overlap Count: 2
# Overlapping Files: accessible/base/nsCoreUtils.cpp, accessible/tests/browser/relations/browser_anchor_positioning.js
# Description:
#   Revert "Bug 1987945: apply code formatting via Lando" for causing bustages at DocAccessible.cpp.
# ==============================================================================

diff -r 826eb4ac00dd -r 09de143215a1 accessible/generic/DocAccessible.cpp
--- a/accessible/generic/DocAccessible.cpp	Tue Sep 30 02:35:22 2025 +0300
+++ b/accessible/generic/DocAccessible.cpp	Tue Sep 30 02:48:50 2025 +0300
@@ -464,27 +464,6 @@
     }
     QueueCacheUpdate(relatedAcc, CacheDomain::Relations);
   }
-
-  if (nsIFrame* anchorFrame = nsCoreUtils::GetAnchorForPositionedFrame(
-          mPresShell, aAcc->GetFrame())) {
-    // If this accessible is anchored, retrieve the anchor and update its
-    // relations.
-    if (LocalAccessible* anchorAcc = GetAccessible(anchorFrame->GetContent())) {
-      if (!mInsertedAccessibles.Contains(anchorAcc)) {
-        QueueCacheUpdate(anchorAcc, CacheDomain::Relations);
-      }
-    }
-  }
-
-  if (nsIFrame* positionedFrame = nsCoreUtils::GetPositionedFrameForAnchor(
-          mPresShell, aAcc->GetFrame())) {
-    // If this accessible is an anchor, retrieve the positioned frame and
-    // refresh the cache on all its anchors.
-    if (LocalAccessible* targetAcc =
-            GetAccessible(positionedFrame->GetContent())) {
-      RefreshAnchorRelationCacheForTarget(targetAcc);
-    }
-  }
 }
 
 ////////////////////////////////////////////////////////////////////////////////
@@ -3182,26 +3161,3 @@
   mAnchorJumpElm = nullptr;
   return true;
 }
-
-void DocAccessible::RefreshAnchorRelationCacheForTarget(
-    LocalAccessible* aTarget) {
-  nsIFrame* frame = aTarget->GetFrame();
-  if (!frame || !frame->HasProperty(nsIFrame::AnchorPosReferences())) {
-    return;
-  }
-
-  AnchorPosReferenceData* referencedAnchors =
-      frame->GetProperty(nsIFrame::AnchorPosReferences());
-  for (auto& entry : *referencedAnchors) {
-    const auto& anchorName = entry.GetKey();
-    if (const nsIFrame* anchorFrame =
-            mPresShell->GetAnchorPosAnchor(anchorName, frame)) {
-      if (LocalAccessible* anchorAcc =
-              GetAccessible(anchorFrame->GetContent())) {
-        if (!mInsertedAccessibles.Contains(anchorAcc)) {
-          QueueCacheUpdate(anchorAcc, CacheDomain::Relations);
-        }
-      }
-    }
-  }
-}